python/ray/tune/trial_executor.py

```diff
@@ -174,10 +192,33 @@ def on_step_end(self, trial_runner):
     def force_reconcilation_on_next_step_end(self):
         pass
 
+    def may_warn_insufficient_resource(self, all_trials):
+        autoscaler_enabled = is_ray_cluster()
+        if not any(trial.status == Trial.RUNNING for trial in all_trials):
+            if self._no_running_trials_since == -1:
+                self._no_running_trials_since = time.monotonic()
+            elif time.monotonic(
+            ) - self._no_running_trials_since > get_warn_threshold(
+                    autoscaler_enabled):
+                warn_prefix = ("If autoscaler is still scaling up, ignore " \
+                              "this message. " if autoscaler_enabled
+                               else "Autoscaler is disabled. ")
+                logger.warn(
+                    warn_prefix +
+                    "Resource is not ready after extended amount of time "
```

Comment:
F-strings are preferable
```suggestion
                    f"{warn_prefix}Resource is not ready after extended amount of time "
```

Reply:
Thanks! Changed to multi-line f-string.